### DISCUSSION TITLE
Design System integration for external libraries

### DISCUSSION THREAD

**Discussion by mmailaender** [formaat-design/reshaped#143] [Ideas]
Reshaped could become the central one-stop shop to create a complete business application. Don't worry about setting up and integrating a consistent design system - Including Basic components and best in class 3rd party components.

Challenges:
- How to ensure we don't bundle all of this together for build setups that don't support tree-shaking. For the current library, even using the whole bundle is fine, while shipping charts / WYSIWYG / code editors and similar features can be a massive overhead in some cases

- This opens a can of worms in terms of support where a lot of issues might be coming from outside our system, so we should still evaluate how much time it might take to maintain these components.

With the rise of Reshaped, we could motivate the library owner to maintain the integrations. If Reshaped is popular enough, it would create a win/win situation for the library owner, Reshaped, and the customer.
Reshaped and 3rd party library: cross-marketing and USP
Customer: First class consistent design system with minimal efforts

---

Using this space to extend it with interesting 3rd party sections continuously:

- Calendar/Scheduler
- Diagrams (e.g. [Reactflow](https://reactflow.dev/))
- Charts (e.g. [Recharts](https://github.com/recharts/recharts))
- Data grid / Tables
- Rich text editor (e.g. [Plate](https://plate.udecode.io/) (Based on [Slate](https://github.com/ianstormtaylor/slate), but an easier-to-use abstraction layer))
- Code Editor
- Billing & Payment (e.g. Stripe)
- Auth (e.g. Clerk.dev)

**Comment by blvdmitry** [comment]
We've decided to only handle the integration docs. We now have issues opened for some of the use cases like this and if someone has any specific requests – feel free to open an issue for the integration guide you want to see 🙏
